### PR TITLE
Reading node instance ID from node spec provider ID - removing depend…

### PR DESCRIPTION
**Pull request by GunaKKIBM**
…ency on node label updater

### CODE REVIEW

File: pkg/metadata/metadata.go [hunk 1]
```diff
@@ -85,11 +84,17 @@ func (nodeManager *NodeInfoManager) NewNodeMetadata(logger *zap.Logger) (NodeMet
 	}
 
 	var workerID string
-	// In case of unmanaged cluster use label NodeInstanceIDLabel for workerID.
-	if os.Getenv(strings.ToUpper("IKS_ENABLED")) != "True" {
+
+	// If the cluster is satellite, the machin-type label equals to UPI
+	// For a satellite cluster, we fetch the workerID from instance ID labals, which is updated by vpc-node-label-updater (init container)
```
Comment (ambiknai) [inline pkg/metadata/metadata.go:89]: *typo **labels**

File: pkg/utils/constants.go [hunk 2]
```diff
@@ -64,6 +64,12 @@ const (
 	// NodeInstanceIDLabel VPC ID label attached to satellite host
 	NodeInstanceIDLabel = "ibm-cloud.kubernetes.io/vpc-instance-id"
 
+	// MachineTypeLabel is a node label is always seen for instances in satellite cluster
```
Comment (ambiknai) [inline pkg/utils/constants.go:67]: MachineTypeLabel is a node label always seen for instances in satellite cluster (Remove extra "is")

File: pkg/metadata/metadata.go [hunk 3]
```diff
@@ -110,3 +115,20 @@ func (manager *nodeMetadataManager) GetRegion() string {
 func (manager *nodeMetadataManager) GetWorkerID() string {
 	return manager.workerID
 }
+
+// fetchInstanceID fetches instance ID of the node from the node provider ID.
+func fetchInstanceID(providerID string) string {
```
Comment (skorati) [inline pkg/metadata/metadata.go:120]: @GunaKKIBM can we reduce the number of splits.

File: pkg/utils/constants.go (hunk 2 above)
Comment (skorati) [reply pkg/utils/constants.go:67]: machine-type is not specific to satellite, just mention that label is used to identify the flavor of worker.

File: pkg/metadata/metadata.go (hunk 1 above)
Comment (skorati) [reply pkg/metadata/metadata.go:89]: can you please change the tone of the speech for these comments.

**Review by skorati — CHANGES_REQUESTED** [review]
looks good, just address these comments and I will approve.

File: pkg/metadata/metadata.go [hunk 4]
```diff
@@ -110,3 +115,20 @@ func (manager *nodeMetadataManager) GetRegion() string {
...
+	// ProviderID looks like: ibm://c468d88aqwerytytytfe0f379bf9///qwertyvvnrja55g/<instance-id>
```
Comment (prankulmahajan) [inline pkg/metadata/metadata.go:121]: Guna can we remove this this, and just keep 
// ProviderID looks like: ibm://<account-id>///<clusterID>/<Instance-ID>

File: pkg/metadata/metadata.go [hunk 5]
```diff
@@ -85,11 +84,17 @@ func (nodeManager *NodeInfoManager) NewNodeMetadata(logger *zap.Logger) (NodeMet
...
+	// If the cluster is satellite, the machin-type label equals to UPI
+	// For a satellite cluster, we fetch the workerID from instance ID labals, which is updated by vpc-node-label-updater (init container)
+	// For managed and IPI cluster, we fetch the workerID from ProviderID in node spec.
+	if nodeLabels[utils.MachineTypeLabel] == utils.UPI {
 		workerID = nodeLabels[utils.NodeInstanceIDLabel]
 	} else {
-		workerID = nodeLabels[utils.NodeWorkerIDLabel]
+		workerID = fetchInstanceID(node.Spec.ProviderID)
+		if workerID == "" {
+			return nil, fmt.Errorf("Unable to fetch instance ID, node provider ID - %s", node.Spec.ProviderID)
```
Comment (prankulmahajan) [inline pkg/metadata/metadata.go:96]:
```suggestion
			return nil, fmt.Errorf("Unable to fetch instance ID for node provider ID - %s", node.Spec.ProviderID)
```

**Review by prankulmahajan — COMMENTED** [review]
Minor suggestions. Logic looks good

File: pkg/metadata/metadata.go (hunk 3 above)
Comment (prankulmahajan) [inline pkg/metadata/metadata.go:120]:
```suggestion
	// ProviderID looks like: ibm://<account-id>///<clusterID>/<Instance-ID>
```

**Review by prankulmahajan — APPROVED** [review]
lgtm

**Review by ambiknai — APPROVED** [review]
lgtm

**Review by skorati — APPROVED** [review]
looks good.

**Review by ambiknai — APPROVED** [review]
lgtm

**Review by prankulmahajan — APPROVED** [review]

**Pull request merged**